src/builder.rs

```diff
@@ -157,42 +219,56 @@ impl SpendInfo {
     }
 }
 
-/// Information about a specific recipient to receive funds in an [`Action`].
+/// Information about a specific output to receive funds in an [`Action`].
 #[derive(Debug)]
-struct RecipientInfo {
+pub struct OutputInfo {
     ovk: Option<OutgoingViewingKey>,
     recipient: Address,
     value: NoteValue,
-    memo: Option<[u8; 512]>,
+    memo: [u8; 512],
 }
 
-impl RecipientInfo {
+impl OutputInfo {
+    /// Constructs a new OutputInfo from its constituent parts.
+    pub fn new(
+        ovk: Option<OutgoingViewingKey>,
+        recipient: Address,
+        value: NoteValue,
+        memo: Option<[u8; 512]>,
+    ) -> Self {
+        Self {
+            ovk,
+            recipient,
+            value,
+            memo: memo.unwrap_or_else(|| {
+                let mut memo = [0; 512];
+                memo[0] = 0xf6;
+                memo
+            }),
+        }
+    }
+
     /// Defined in [Zcash Protocol Spec § 4.8.3: Dummy Notes (Orchard)][orcharddummynotes].
     ///
     /// [orcharddummynotes]: https://zips.z.cash/protocol/nu5.pdf#orcharddummynotes
-    fn dummy(rng: &mut impl RngCore) -> Self {
+    pub fn dummy(rng: &mut impl RngCore) -> Self {
```

Comment:
Why does this need to be public? `SpendInfo::dummy` is not public (so there's an API asymmetry now), and the builder is responsible for applying padding internally.
```suggestion
    fn dummy(rng: &mut impl RngCore) -> Self {
```

Reply:
I want it to be possible for callers to add dummy actions beyond the two required padding; they can already do so by manually sending zero value to a random address, so exposing this convenience method doesn't add any capability that they don't already have, and I don't see there as being a substantive maintenance burden.